[BUG] zdnn.h does not have name mangling disabled for C++

<!--
Note: Please use the search to see if a similar issue already exists.
-->

### Current Observation:
Currently when compiling C++ code with references to `zdnn.h`, the linker will fail to find zDNN references because C++ is doing name mangling. An example of the ambiguous error during linking is shown below.

```
/usr/bin/ld: ../../bin/libggml-zdnn.so: undefined reference to `zdnn_transform_ztensor(zdnn_ztensor*, ...)'
/usr/bin/ld: ../../bin/libggml-zdnn.so: undefined reference to `zdnn_init_ztensor_with_malloc(zdnn_tensor_desc*, zdnn_tensor_desc*, zdnn_ztensor*)'
/usr/bin/ld: ../../bin/libggml-zdnn.so: undefined reference to `zdnn_transform_origtensor(zdnn_ztensor const*, void*)'
/usr/bin/ld: ../../bin/libggml-zdnn.so: undefined reference to `zdnn_init_pre_transformed_desc(zdnn_data_layouts, zdnn_data_types, zdnn_tensor_desc*, ...)'
/usr/bin/ld: ../../bin/libggml-zdnn.so: undefined reference to `zdnn_free_ztensor_buffer(zdnn_ztensor const*)'
/usr/bin/ld: ../../bin/libggml-zdnn.so: undefined reference to `zdnn_add(zdnn_ztensor const*, zdnn_ztensor const*, zdnn_ztensor*)'
/usr/bin/ld: ../../bin/libggml-zdnn.so: undefined reference to `zdnn_generate_transformed_desc(zdnn_tensor_desc const*, zdnn_tensor_desc*)'
```

### Expected:
<!-- A concise description of what you expected to see. -->
Compiling for C and C++ with references to `zdnn.h` should have the linker successfully find the references and link.

### Location:
<!-- Provide some details to the location. -->
Affected file is `zdnn.h`

### Anything else:
<!--
Links? References? Anything that will give more context about the issue!
-->
Will file a PR to fix this issue

--- a/zdnn/zdnn.h
+++ b/zdnn/zdnn.h
@@ -23,6 +23,10 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 // -----------------------------------------------------------------------------
 // NOTE:
 // Ensure that symbols in zdnn.h and zdnn.map are in sync!
@@ -649,5 +653,8 @@ zdnn_status zdnn_get_max_limit(zdnn_data_types transformed_type,
 zdnn_status zdnn_get_min_limit(zdnn_data_types transformed_type,
                                zdnn_data_types pre_transformed_type,
                                void *limit);
+#ifdef __cplusplus
+}
+#endif /* __cplusplus */
 
 #endif /* ZDNN_ZDNN_H_ */

